blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java

```diff
@@ -76,11 +76,11 @@ public String getScmId() {
     public String getApiUrl(@Nonnull Item item) {
         if (item instanceof org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject) {
             MultiBranchProject<?,?> mbp = (MultiBranchProject<?,?>)item;
-            SCMSource s = mbp.getSCMSources().iterator().next();
```

Comment:
This blows up if `iterator().hasNext()` is false.

Reply:
maybe instead, then just do
```
Iterator<SCMSource> i = mbp.getSCMSources();
if (i.hasNext()) {
...
```

seems a bit overkill to add some closures in to this logic

Reply:
@kzantow why is this overkill? I could do that ugly if block but this looks simple enough:)

Reply:
Efficiency reasons?

Reply:
And, to me, Java streaming stuff is less clear than imperative style programming, but I'll admit that's because I'm old :)

Reply:
Matter of style I guess. I did few projects in past and try to write new code with functional syntax where possible. Efficiency wise stream parallels can have perf issues compared to sequential but I see mixed report for that. In this particular case it’s harmless and we are discussing style mostly.

Reply:
yes, mostly a matter of style, not a frequently called method. :)